### PR TITLE
Historical analog retrieval encoder + /analyze/analogs endpoint (#294)

### PR DESCRIPTION
**Pull request by yusufizzetmurat**
## Summary

- Fine-tune driver `app/retrieval/train.py` — sentence-transformer MNRL contrastive layer over `finbert_fed_adjacent_xbank_dapt`, same-meeting (statement, minutes / press-conference) positive pairs.
- Index module `app/retrieval/index.py` — embeds every historical FOMC statement from `events.parquet`, persists `index.parquet` + `embeddings.npy` + `manifest.json`, plain numpy cosine top-k for the ~250-row corpus.
- New `POST /analyze/analogs` endpoint backed by a thread-safe singleton in `app/services/analogs.py`; degrades to empty when the bundle is absent.
- Registers `finbert_fed_adjacent_xbank_dapt_retrieval` alias; checkpoint stays local until the deployability lane (#302) mirrors it to Hub.
- `sentence-transformers` added to backend deps; inference uses plain `AutoTokenizer`/`AutoModel` + masked mean-pool.

## Test plan

- `docker compose run --rm backend pytest tests/unit/test_retrieval_index.py tests/unit/test_analogs_endpoint.py -q`
- `docker compose run --rm backend pytest tests/unit/test_main_api.py tests/unit/test_analyze_persists_history.py tests/unit/test_research_endpoints.py -q`
- `docker compose run --rm backend ruff check app/retrieval app/services/analogs.py app/schemas.py`

### CODE REVIEW

**Comment by yusufizzetmurat** [comment]
Pushed `bbd9e32` to address the review findings:

- Drop `forward_realized_vol_10d` from `AnalogCard`; surface a coarse `subsequent_vol_regime` bucket (`calm`/`normal`/`high`) pinned to a held-out 2000-2015 reference distribution so the API output is not a label leak.
- Add `--train-end` / `--fold-id` walk-forward filter to `app.retrieval.train`; rows >= cutoff are dropped before pair construction; resolved boundary persists into `manifest.json`.
- `as_of_date` strict-backward filter added to `/analyze/analogs`, `find_analogs`, and the underlying `query`; returns what's eligible (no pad with future rows).
- Self-match suppression: query path drops the row whose `text_hash` matches the cleaned query's sha256; removed the degenerate (statement, statement) self-pair from the pair builder.
- Atomic bundle writes via sibling .tmp + `os.replace`; manifest written last so the bundle is either complete or absent. `np.load` now passes `allow_pickle=False`.
- `AutoTokenizer` / `AutoModel` load with `local_files_only=True, trust_remote_code=False` so a missing checkpoint never silently triggers a HF Hub fetch.
- `bundle_available()` wired as endpoint pre-flight; unexpected-exception arm logs server-side and returns sanitised `detail="Analog retrieval unavailable"` (no internal paths).
- Query path truncates to `MAX_TEXT_CHARS` before tokenisation; `fine_tune_and_index` rejects `batch_size < 2` for MNRL; explicit `torch.Generator().manual_seed(seed)` on the training DataLoader.
- New tests: `test_retrieval_train_pairs.py` (filter / sibling / train_end coverage); `test_analogs_endpoint.py` gains 503-sanitisation + `as_of_date` filter cases; `test_retrieval_index.py` gains self-match suppression + atomic-write + `train_end` manifest checks.

Wiki pages 07 and 16 updated in `fc8a3ad`.
